### ISSUE TITLE
Number of iterations for LearningRateScheduler 

### ISSUE THREAD

**Issue by maksym-del** [allenai/allennlp#4089]
There are 2 issues I see there:
1)```line 1002```: we use ```gpu batches``` as a number of steps per epoch while the correct behavior would be using a number of ```effective batches``` obtained after gradient accumulation.
2) ```line 1006```: there are schedulers that do not accept ```num_steps_per_epoch``` as their constructor argument. Are they handled somehow or will they result in an error?
https://github.com/allenai/allennlp/blob/039522a4a343aef006e5b5fdb15fd67be49f8748/allennlp/training/trainer.py#L1002-L1006

**Comment by dirkgr** [comment]
Great catch for issue 1: https://github.com/allenai/allennlp/pull/4099

Issue 2 is not a problem. The `from_params` magic takes care of it. We have many training configs that use learning rate schedulers other than `slanted_triangular`, and I checked: They work.

**Comment by maksym-del** [comment]
After https://github.com/allenai/allennlp/issues/4044 we will have a bunch of new ```lr_schedulers``` that take ```num_training_steps``` parameter as an input. 

Is it true that we can just do like below?
```python
learning_rate_scheduler_ = learning_rate_scheduler.construct( 
optimizer=optimizer_,
num_epochs=num_epochs, 
num_steps_per_epoch=batches_per_epoch,
num_trainig_steps=batches_per_epoch * num_epochs)

```

**Comment by dirkgr** [comment]
@maksym-del, yes, I think that will work.

Can we standardize on one way of specifying this instead of having two? Are there any LR schedulers that need to know the number of epochs? If not, we could get away with just specifying `num_training_steps`.

**Comment by dirkgr** [comment]
I feel that providing `num_epochs` and `batches_per_epoch` provides strictly more information than `num_training_steps`. So I would rather just pass that, and calculate the extra parameter in the wrapper.
